src/analyze/dependencies.ts

```diff
@@ -1,102 +1,294 @@
+    severity = 'exact';
+  } else {
+    severity = 'conflict';
+  }
+
+  return {
+    name: packageName,
+    versions: nodes,
+    severity,
+    potentialSavings: calculatePotentialSavings(nodes),
+    suggestions: generateSuggestions(nodes)
+  };
+}
+
+/**
+ * Calculates potential savings from deduplication
+ */
+function calculatePotentialSavings(nodes: DependencyNode[]): number {
+  // For now, return a simple estimate based on number of duplicates
+  // TODO: Implement actual size calculation
+  return nodes.length - 1;
+}
+
+/**
+ * Generates suggestions for resolving duplicates
+ */
+function generateSuggestions(nodes: DependencyNode[]): string[] {
+  const suggestions: string[] = [];
+
+  // Group by version to identify the most common version
+  const versionCounts = new Map<string, number>();
+  for (const node of nodes) {
+    versionCounts.set(
+      node.version,
+      (versionCounts.get(node.version) || 0) + 1
+    );
+  }
+
+  const mostCommonVersion = Array.from(versionCounts.entries()).sort(
+    (a, b) => b[1] - a[1]
+  )[0];
+
+  if (mostCommonVersion && mostCommonVersion[1] > 1) {
+    suggestions.push(
+      `Consider standardizing on version ${mostCommonVersion[0]} (used by ${mostCommonVersion[1]} dependencies)`
+    );
+  }
+
+  // Suggest checking for newer versions of consuming packages
+  const uniqueParents = new Set(nodes.map((n) => n.parent).filter(Boolean));
+  if (uniqueParents.size > 1) {
+    suggestions.push(
+      `Check if newer versions of consuming packages (${Array.from(uniqueParents).join(', ')}) would resolve this duplicate`
+    );
+  }
+
+  return suggestions;
+}
+
 // Keep the existing tarball analysis for backward compatibility
 export async function analyzeDependencies(
   fileSystem: FileSystem
 ): Promise<DependencyStats> {
   const packageFiles = await fileSystem.listPackageFiles();
   const rootDir = await fileSystem.getRootDir();
 
-  // Debug: Log all files in the tarball
-  logger.debug('Package.json files:');
-  for (const file of packageFiles) {
-    logger.debug(`- ${file}`);
-  }
-
-  // Find package.json
+  // Find root package.json
   let pkg: PackageJsonLike;
-
   try {
     pkg = JSON.parse(await fileSystem.readFile(rootDir + '/package.json'));
   } catch {
     throw new Error('No package.json found.');
   }
 
-  // Calculate total size
   const installSize = await fileSystem.getInstallSize();
-
-  // Count dependencies
   const directDependencies = Object.keys(pkg.dependencies || {}).length;
   const devDependencies = Object.keys(pkg.devDependencies || {}).length;
 
-  // Count CJS vs ESM dependencies
   let cjsDependencies = 0;
   let esmDependencies = 0;
+  const dependencyNodes: DependencyNode[] = [];
 
-  // Analyze each dependency
-  for (const file of packageFiles) {
-    if (file === rootDir + '/package.json') {
-      continue;
+  // Helper to parse a package.json file
+  async function parsePackageJson(path: string): Promise<PackageJsonLike | null> {
```

Comment:
this doesn't need to be inside `analyzeDependencies`, i.e. it can be hoisted out